drivers/input/touchscreen/rpi-ft5406.c

```diff
@@ -81,18 +84,21 @@ static int ft5406_thread(void *arg)
 				int x = (((int) regs.point[i].xh & 0xf) << 8) + regs.point[i].xl;
 				int y = (((int) regs.point[i].yh & 0xf) << 8) + regs.point[i].yl;
 				int touchid = (regs.point[i].yh >> 4) & 0xf;
```

Comment:
Is this value going to be correct?  Looks like you've got 2 bits for touchid and 2 bits for touch number...  But this is supposed to be a five touch device, in fact I believe the old one was 10 point...